src/dodal/devices/oav/pin_image_recognition/__init__.py

```diff
@@ -1,46 +1,44 @@
 import asyncio
 import time
 from collections import OrderedDict
-from typing import Callable, Optional, Tuple, Type, TypeVar
+from typing import Optional, Tuple, Type, TypeVar
 
 import numpy as np
 from bluesky.protocols import Descriptor
 from numpy.typing import NDArray
-from ophyd.v2.core import Device, Readable, Reading, SimConverter, SimSignalBackend
+from ophyd.v2.core import Device, Readable, Reading, SimSignalBackend
 from ophyd.v2.epics import SignalR, SignalRW, epics_signal_r
 
-from dodal.devices.oav.pin_tip_detection.pin_tip_detect_utils import (
-    ArrayProcessingFunctions,
+from dodal.devices.oav.pin_image_recognition.utils import (
+    ARRAY_PROCESSING_FUNCTIONS_MAP,
     MxSampleDetect,
+    ScanDirections,
+    identity,
 )
 from dodal.log import LOGGER
 
 T = TypeVar("T")
 
 
-def _create_soft_signal(
-    name: str, datatype: T | Type[T], initial_value: T
-) -> SignalRW[T]:
-    class _Converter(SimConverter):
-        def make_initial_value(self, *args, **kwargs) -> T:
-            return initial_value
+def _create_soft_signal(datatype: Type[T], name: str) -> SignalRW[T]:
+    return SignalRW(
+        SimSignalBackend(
+            datatype, f"sim://oav_with_pin_tip_detection_soft_params:{name}"
+        )
+    )
 
-    class _SimSignalBackend(SimSignalBackend):
-        async def connect(self) -> None:
-            self.converter = _Converter()
-            self._initial_value = self.converter.make_initial_value()
-            self._severity = 0
 
-            await self.put(None)
+class PinTipDetection(Readable, Device):
+    """
+    A device which will read a single frame from an on-axis view and use that frame
+    to calculate the pin-tip offset (in pixels) of that frame.
 
-    backend = _SimSignalBackend(
-        datatype, f"sim://oav_with_pin_tip_detection_soft_params:{name}"
-    )
-    signal: SignalRW[T] = SignalRW(backend)
-    return signal
+    Used for pin tip centring workflow.
 
+    Note that if the sample is off-screen, this class will return the centre as the "edge"
+    of the image.
```

Comment:
Should: Is this true? I thought it returned -1 if the sample was completely offscreen and the edge if the sample extended across the whole screen?

Reply:
Have clarified comment, I think it's just potential ambiguity between "sample is off-screen" and "tip of the sample is off-screen". Hopefully new wording is clearer.